pallets/author-slot-filter/src/lib.rs

```diff
@@ -34,9 +34,54 @@ mod benchmarks;
 
 pub mod weights;
 
+use frame_support::{Deserialize, Serialize};
+use parity_scale_codec::{Decode, Encode};
+use scale_info::TypeInfo;
+
+#[derive(
+	Clone,
+	Debug,
+	TypeInfo,
+	Serialize,
+	Deserialize,
+	Encode,
+	Decode,
+	PartialEq,
+	Eq,
+	PartialOrd,
+	Ord,
+	Hash,
+)]
+pub struct NonZeroU32(u32);
+
+impl core::ops::Deref for NonZeroU32 {
+	type Target = u32;
+	fn deref(&self) -> &Self::Target {
+		&self.0
+	}
+}
+
+impl parity_scale_codec::EncodeLike<u32> for NonZeroU32 {}
```

Comment:
Since an instance of your type `NonZeroU32` can be created by SCALE decoding, this type can be 0, so we don't have the desired guarantee. 
Instead, `Decode` should be implemented by hand to return an error if the storage contains zero.